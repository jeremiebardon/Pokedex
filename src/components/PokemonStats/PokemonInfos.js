import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'antd';

import getPokemonChain from '../../helpers/index';

export const PokemonMoves = ({ moves }) => (
  <div>
    <Table
      columns={[
        {
          title: 'All Moves',
          dataIndex: 'move',
          key: 'move',
          render: (move) => move.name,
        },
      ]}
      pagination={{ position: ['', 'bottomCenter'] }}
      dataSource={moves}
      rowKey={(record) => `move-${record.move.name}`}
    />
  </div>
);

export const PokemonStats = ({ stats }) => (
  <div>
    <Table
      columns={[
        {
          title: 'Stat',
          dataIndex: 'stat',
          key: 'stat',
          render: (stat) => <p key={stat.name}>{stat.name}</p>,
        },
        {
          title: 'Value',
          dataIndex: 'base_stat',
          key: 'base_stat',
          render: (baseStats) => <p key={baseStats}>{baseStats}</p>,
        },
      ]}
      dataSource={stats}
      pagination={false}
      rowKey={(record) => `stat-${record.stat.name}`}
    />
  </div>
);

export const PokemonEvolution = ({ evolution, currentPokemon, changePokemon }) => (
  <Table
    columns={[
      {
        title: 'Name',
        dataIndex: 'name',
        key: 'name',
        render: (name) => name,
      },
      {
        title: 'Level',
        dataIndex: 'min_level',
        key: currentPokemon,
        render: (baseStats) => baseStats,
      },
      {
        title: 'Stage',
        dataIndex: 'depth',
        key: currentPokemon,
        render: (baseStats) => baseStats,
      },
    ]}
    onRow={(record) => {
      return {
        onClick: () => changePokemon(record.name),
      };
    }}
    rowClassName={(record) => (
      `${record.name === currentPokemon ? 'current' : ''} selectable-row`
    )}
    dataSource={getPokemonChain(evolution.chain)}
    pagination={false}
    rowKey="name"
  />
);

PokemonEvolution.propTypes = {
  evolution: PropTypes.shape({
    baby_trigger_item: PropTypes.bool,
    chain: PropTypes.object.isRequired,
    id: PropTypes.number.isRequired,
  }).isRequired,
  currentPokemon: PropTypes.string.isRequired,
  changePokemon: PropTypes.func.isRequired,
};

PokemonStats.propTypes = {
  stats: PropTypes.arrayOf(
    PropTypes.shape({
      base_stat: PropTypes.number,
      effort: PropTypes.number,
      stat: PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
      }),
    }),
  ).isRequired,
};

PokemonMoves.propTypes = {
  moves: PropTypes.arrayOf(
    PropTypes.shape({
      move: PropTypes.object,
      version_group_details: PropTypes.arrayOf(
        PropTypes.shape({
          level_learned_at: PropTypes.number,
          move_learn_method: PropTypes.object,
          version_group: PropTypes.object,
        }),
      ),
    }),
  ).isRequired,
};
