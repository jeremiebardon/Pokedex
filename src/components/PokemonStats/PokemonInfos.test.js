import React from 'react';
import '../../../__mocks__/matchMedia';
import { render, fireEvent } from '@testing-library/react';
import { test, expect, jest } from 'jest-without-globals';

import { PokemonEvolution, PokemonStats, PokemonMoves } from './PokemonInfos';

const evolutionProps = {
  evolution: {
    baby_trigger_item: null,
    chain: {
      evolution_details: [],
      evolves_to: [
        {
          evolution_details: [
            {
              gender: null,
              held_item: null,
              item: null,
              known_move: null,
              known_move_type: null,
              location: null,
              min_affection: null,
              min_beauty: null,
              min_happiness: null,
              min_level: 16,
              needs_overworld_rain: false,
              party_species: null,
              party_type: null,
              relative_physical_stats: null,
              trade_species: null,
              trigger: {
                name: 'level-up',
                url: 'https://pokeapi.co/api/v2/evolution-trigger/1/',
              },
              turn_upside_down: false,
            },
          ],
          evolves_to: [
            {
              evolution_details: [
                {
                  gender: null,
                  held_item: null,
                  item: null,
                  known_move: null,
                  known_move_type: null,
                  location: null,
                  min_affection: null,
                  min_beauty: null,
                  min_happiness: null,
                  min_level: 32,
                  needs_overworld_rain: false,
                  party_species: null,
                  party_type: null,
                  relative_physical_stats: null,
                  trade_species: null,
                  trigger: {
                    name: 'level-up',
                    url: 'https://pokeapi.co/api/v2/evolution-trigger/1/',
                  },
                  turn_upside_down: false,
                },
              ],
              evolves_to: [],
              is_baby: false,
              species: {
                name: 'venusaur',
                url: 'https://pokeapi.co/api/v2/pokemon-species/3/',
              },
            },
          ],
          is_baby: false,
          species: {
            name: 'ivysaur',
            url: 'https://pokeapi.co/api/v2/pokemon-species/2/',
          },
        },
      ],
      is_baby: false,
      species: {
        name: 'bulbasaur',
        url: 'https://pokeapi.co/api/v2/pokemon-species/1/',
      },
    },
    id: 1,
  },
  currentPokemon: 'bulbasaur',
  changePokemon: jest.fn(),
  getPokemonChain: jest.fn(),
};

const statsProps = {
  stats: [
    {
      base_stat: 45,
      effort: 0,
      stat: {
        name: 'hp',
        url: 'https://pokeapi.co/api/v2/stat/1/',
      },
    },
    {
      base_stat: 49,
      effort: 0,
      stat: {
        name: 'attack',
        url: 'https://pokeapi.co/api/v2/stat/2/',
      },
    },
    {
      base_stat: 49,
      effort: 0,
      stat: {
        name: 'defense',
        url: 'https://pokeapi.co/api/v2/stat/3/',
      },
    },
    {
      base_stat: 65,
      effort: 1,
      stat: {
        name: 'special-attack',
        url: 'https://pokeapi.co/api/v2/stat/4/',
      },
    },
    {
      base_stat: 65,
      effort: 0,
      stat: {
        name: 'special-defense',
        url: 'https://pokeapi.co/api/v2/stat/5/',
      },
    },
    {
      base_stat: 45,
      effort: 0,
      stat: {
        name: 'speed',
        url: 'https://pokeapi.co/api/v2/stat/6/',
      },
    },
  ],
};

const movesProps = {
  moves: [
    {
      move: {
        name: 'razor-wind',
        url: 'https://pokeapi.co/api/v2/move/13/',
      },
      version_group_details: [
        {
          level_learned_at: 0,
          move_learn_method: {
            name: 'egg',
            url: 'https://pokeapi.co/api/v2/move-learn-method/2/',
          },
          version_group: {
            name: 'gold-silver',
            url: 'https://pokeapi.co/api/v2/version-group/3/',
          },
        },
        {
          level_learned_at: 0,
          move_learn_method: {
            name: 'egg',
            url: 'https://pokeapi.co/api/v2/move-learn-method/2/',
          },
          version_group: {
            name: 'crystal',
            url: 'https://pokeapi.co/api/v2/version-group/4/',
          },
        },
      ],
    },
  ],
};

/* eslint-disable */
test('Render PokemonEvolution', () => {
  const component = render(<PokemonEvolution {...evolutionProps}/>);
  expect(component).toMatchSnapshot();
});

test('Render PokemonStats', () => {
  const component = render(<PokemonStats {...statsProps}/>);
  expect(component).toMatchSnapshot();
});

test('Render PokemonMoves', () => {
  const component = render(<PokemonMoves {...movesProps}/>);
  expect(component).toMatchSnapshot();
});

test('Pokemon Evolution Row Click Change Page', () => {
  const { container } = render(<PokemonEvolution {...evolutionProps}/>);
  const element = container.querySelectorAll('.ant-table-row')[1];
  fireEvent.click(element, container);
  expect(evolutionProps.changePokemon).toHaveBeenCalledWith('ivysaur');
});
