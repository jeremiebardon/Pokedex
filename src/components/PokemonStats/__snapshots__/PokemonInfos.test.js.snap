// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PokemonEvolution 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Name
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Level
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Stage
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0 current selectable-row"
                        data-row-key="bulbasaur"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          bulbasaur
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          1
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0  selectable-row"
                        data-row-key="ivysaur"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          ivysaur
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          16
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0  selectable-row"
                        data-row-key="venusaur"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          venusaur
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          32
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          3
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-table-wrapper"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup />
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                      >
                        Name
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Level
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Stage
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0 current selectable-row"
                      data-row-key="bulbasaur"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        bulbasaur
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        0
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        1
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0  selectable-row"
                      data-row-key="ivysaur"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        ivysaur
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        16
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        2
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0  selectable-row"
                      data-row-key="venusaur"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        venusaur
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        3
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render PokemonMoves 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                          >
                            All Moves
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="move-razor-wind"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            razor-wind
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination ant-table-pagination-center"
                unselectable="unselectable"
              >
                <li
                  aria-disabled="true"
                  class="ant-pagination-prev ant-pagination-disabled"
                  title="Previous Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="left"
                      class="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a
                    rel="nofollow"
                  >
                    1
                  </a>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-next ant-pagination-disabled"
                  title="Next Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="right"
                      class="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          All Moves
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="move-razor-wind"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          razor-wind
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-center"
              unselectable="unselectable"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render PokemonStats 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                          >
                            Stat
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Value
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="stat-hp"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              hp
                            </p>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              45
                            </p>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="stat-attack"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              attack
                            </p>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              49
                            </p>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="stat-defense"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              defense
                            </p>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              49
                            </p>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="stat-special-attack"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              special-attack
                            </p>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              65
                            </p>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="stat-special-defense"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              special-defense
                            </p>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              65
                            </p>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="stat-speed"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              speed
                            </p>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <p>
                              45
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Stat
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Value
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="stat-hp"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            hp
                          </p>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            45
                          </p>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="stat-attack"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            attack
                          </p>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            49
                          </p>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="stat-defense"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            defense
                          </p>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            49
                          </p>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="stat-special-attack"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            special-attack
                          </p>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            65
                          </p>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="stat-special-defense"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            special-defense
                          </p>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            65
                          </p>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="stat-speed"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            speed
                          </p>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <p>
                            45
                          </p>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
